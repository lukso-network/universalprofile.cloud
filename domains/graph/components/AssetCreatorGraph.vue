<script setup lang="ts">
type Props = {
  creator?: Creator
  asset?: Asset
}

const props = defineProps<Props>()
const { isMobile } = useDevice()
const { formatMessage } = useIntl()

const handleOpenProfile = (address?: Address) => {
  if (address) {
    navigateTo(profileRoute(address))
  }
}

const verified = computed(() => {
  const issued = useIssuedAssetsGraph().validateAssets(
    props.creator ? [props.creator] : [],
    props.asset?.address
  )
  return (
    (props.creator?.address && issued.value?.get(props.creator?.address)) ||
    false
  )
})

const profileImage = useProfileAvatar(props.creator, 24)
</script>

<template>
  <lukso-button
    v-if="creator"
    custom-class="px-4"
    variant="secondary"
    :is-full-width="isMobile ? true : undefined"
    @click="handleOpenProfile(creator?.address)"
  >
    <div class="flex w-full items-center justify-between">
      <div class="flex items-center">
        <lukso-profile
          size="x-small"
          :profile-url="profileImage?.url"
          :profile-address="creator?.address"
          has-identicon
        ></lukso-profile>
        <lukso-username
          :name="creator?.name"
          :address="creator?.address"
          size="small"
          class="pl-2"
          name-color="neutral-20"
        ></lukso-username>
      </div>
      <div class="flex items-center">
        <lukso-tooltip
          v-if="verified"
          variant="success"
          :text="formatMessage('asset_creators_verified')"
          class="ml-2"
        >
          <lukso-icon
            name="complete-filled"
            color="green-54"
            secondary-color="neutral-100"
            size="small"
          ></lukso-icon>
        </lukso-tooltip>
        <lukso-tooltip
          v-else
          variant="danger"
          :text="formatMessage('asset_creators_unverified')"
          class="ml-2"
        >
          <lukso-icon
            name="cross-filled"
            color="red-55"
            secondary-color="neutral-100"
            size="small"
          ></lukso-icon>
        </lukso-tooltip>
      </div>
    </div>
  </lukso-button>
</template>
