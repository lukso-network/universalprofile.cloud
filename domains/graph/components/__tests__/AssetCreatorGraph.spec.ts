import { renderSuspended } from '@nuxt/test-utils/runtime'
import { beforeEach, describe, expect, it } from 'vitest'

import AssetCreatorGraph from '@/domains/graph/components/AssetCreatorGraph.vue'

describe('AssetCreatorGraph', () => {
  beforeEach(() => {
    useIntl().setupIntl(defaultConfig)
  })

  it('should render empty state', async () => {
    const component = await renderSuspended(AssetCreatorGraph)

    expect(component.html()).toMatchSnapshot()
  })

  it('should render unverified creator', async () => {
    const component = await renderSuspended(AssetCreatorGraph, {
      props: {
        creator: {
          address: '0x123',
          name: 'John Doe',
          issuedAssets: ['0x456'],
        },
        asset: {},
      },
    })

    expect(component.html()).toMatchSnapshot()
  })

  it('should render verified creator', async () => {
    const component = await renderSuspended(AssetCreatorGraph, {
      props: {
        creator: {
          address: '0x123',
          name: 'John Doe',
          issuedAssets: ['0x456'],
        },
        asset: {
          address: '0x456',
        },
      },
    })

    expect(component.html()).toMatchSnapshot()
  })
})
