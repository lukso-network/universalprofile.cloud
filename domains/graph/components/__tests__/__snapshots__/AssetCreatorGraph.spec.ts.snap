// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`AssetCreatorGraph > should render empty state 1`] = `
"<div id="test-wrapper">
  <!--v-if-->
</div>"
`;

exports[`AssetCreatorGraph > should render unverified creator 1`] = `
"<div id="test-wrapper">
  <lukso-button custom-class="px-4" variant="secondary">
    <div class="flex w-full items-center justify-between">
      <div class="flex items-center">
        <lukso-profile size="x-small" profile-address="0x123" has-identicon=""></lukso-profile>
        <lukso-username name="John Doe" address="0x123" size="small" class="pl-2" name-color="neutral-20"></lukso-username>
      </div>
      <div class="flex items-center">
        <lukso-tooltip variant="danger" text="Un-verified creator" class="ml-2">
          <lukso-icon name="cross-filled" color="red-55" secondary-color="neutral-100" size="small"></lukso-icon>
        </lukso-tooltip>
      </div>
    </div>
  </lukso-button>
</div>"
`;

exports[`AssetCreatorGraph > should render verified creator 1`] = `
"<div id="test-wrapper">
  <lukso-button custom-class="px-4" variant="secondary">
    <div class="flex w-full items-center justify-between">
      <div class="flex items-center">
        <lukso-profile size="x-small" profile-address="0x123" has-identicon=""></lukso-profile>
        <lukso-username name="John Doe" address="0x123" size="small" class="pl-2" name-color="neutral-20"></lukso-username>
      </div>
      <div class="flex items-center">
        <lukso-tooltip variant="success" text="Verified creator" class="ml-2">
          <lukso-icon name="complete-filled" color="green-54" secondary-color="neutral-100" size="small"></lukso-icon>
        </lukso-tooltip>
      </div>
    </div>
  </lukso-button>
</div>"
`;
